agent.go

```diff
@@ -246,10 +247,25 @@ func (c *controller) agentInit(bindAddrOrInterface, advertiseAddr string) error
 	}
 
 	keys, tags := c.getKeys(subsysGossip)
+
+	// Make the hostname unique in the gossip cluster by appending a 12
+	// byte random ID. Between distros there is no consensus if the
+	// hostname should be an FQDN or shortname. Limit the hostname to
+	// RFC 1123 compliant 63 bytes
 	hostname, _ := os.Hostname()
+	id := stringid.TruncateID(stringid.GenerateRandomID())
+
+	var nodeName string
+	if len(hostname)+len(id) > 63 {
```

Comment:
@mrjana @aboch Removed the truncation & verified that on a daemon restart other nodes are marking the previous node name as a suspicious node and eventually marking it failed.